### DISCUSSION TITLE
Flow per body surface area

### DISCUSSION THREAD

**Discussion by DonkeySaddle** [Open-Systems-Pharmacology/Forum#1322] [Questions & Problems] [answered]
I am trying to load a Mobi(pkml) simulation into R and am getting the following error:

"Error in rClr::clrCall(simulationPersister, "LoadSimulation", expandPath(filePath),  : 
  Type:    System.Collections.Generic.KeyNotFoundException
Message: Dimension 'Flow per body surface area' not available in DimensionFactory.
Method:  OSPSuite.Core.Domain.UnitSystem.IDimension Dimension(System.String)
Stack trace:
   at OSPSuite.Core.Domain.UnitSystem.DimensionFactory.Dimension(String name)
   at OSPSuite.Core.Serialization.Xml.SerializationContext.DimensionByName(String dimensionName)
   at OSPSuite.Core.Serialization.Xml.DimensionAttributeMapper.ConvertFrom(String attributeValue, SerializationContext context)
   at OSPSuite.Serializer.Xml.XmlAttributeMapper`2.Deserialize(TObject objectToDeserialize, XElement element, TContext context)
   at OSPSuite.Serializer.Xml.XmlSerializer`2.TypedDeserialize(TObject objectToDeserialize, XElement outputToDeserialize, TContext context)
   at OSPSuite.Core.Serialization.Xml.ObjectBaseXmlSerializer`1.TypedDeserialize(T objectToDeserialize, XElement outputToDeserialize, SerializationContext serializationContext)
   at OSPSui"

I know this issue was mentioned [here](https://github.com/Open-Systems-Pharmacology/MoBi/issues/694) and has since been resolved, but I am not exactly sure what I should do to fix it. Do I need to update my R packages? 

Any assistance would be greatly appreciated. 

Thanks,

DS

**Comment by PavelBal** [comment, marked as answer]
What is the R package version you are using?

**Reply by DonkeySaddle** [reply]
That’s probably what it is. I updated my versions of Pk-Sim & Mobi, but didn’t update the R package. Unfortunately, I’m away and won’t be able to check for some time. 

Thanks @PavelBal!

DS

**Reply by AndreDlm** [reply]
Most likely it's because of the `DimensionFactory` in the R package, but I am not sure if the latest release version has this fixed...
@Yuri05 can probably comment on that. 

If I'm not wrong, the dimension was included when the chronic kidney disease (CKD) population became part of the PK-Sim database, which entailed the implementation of a new parameter called `eGFR` that has the dimension `Flow per body surface area`. If you do not simulate CKD individuals, you don't need this parameter. So a workaround would be to delete the parameter and its formula from the spatial structure (see below) and update your simulation. It should then run in R.

![image](https://user-images.githubusercontent.com/26541603/192963707-c833627c-7009-493a-9cb1-7800d7711315.png)
![image](https://user-images.githubusercontent.com/26541603/192963800-8fad854e-18f5-4060-925a-66272b4106b4.png)

**Reply by DonkeySaddle** [reply]
Ahhh, that makes sense. Thanks for the clear explanation with the corresponding photos. This is great @AndreDlm. 

Will report back when I get a chance to implement. Regardless I should probably update the R package.

**Reply by DonkeySaddle** [reply]
@AndreDlm Hmm,  I am struggling to delete the eGFR parameter and it's formula in MOBI. I can't seem to replicate the screenshots you've posted. Presumably I am doing something stupid...that's generally the case

Anyways, I can see the parameter(but I don't have the option to delete it...):  
<img width="813" alt="image" src="https://user-images.githubusercontent.com/25123758/194776679-240c3883-6bc7-48b4-a48e-7ed24ff591f5.png">


Thanks,

DS

**Reply by Yuri05** [reply]
@DonkeySaddle You cannot delete the parameter directly in the simulation. You must delete it in the used spatial structure first and then update it in the simulation.
Or just update the ospsuite R package to the latest released version...

**Reply by DonkeySaddle** [reply]
@Yuri05 "Or just update the ospsuite R package to the latest released version..." That seems to have done it. Thank you.
